src/main/kotlin/net/irisshaders/lilybot/extensions/config/Config.kt

```diff
@@ -1,188 +1,363 @@
+	}
+
+	unsafeSubCommand(::LoggingArgs) {
+		name = "logging"
+		description = "Configure Lily's logging system"
+
+		initialResponse = InitialSlashCommandResponse.None
 
-		/** The channel for sending join logs to. */
-		val joinChannel by channel {
-			name = "joinChannel"
-			description = "Your Join Logs Channel"
+		check {
+			anyGuild()
+			// hasPermission(Permission.ManageGuild)
 		}
 
-		/** The role for the support team. */
-		val supportTeam by optionalRole {
-			name = "supportTeamRole"
-			description = "Your Support Team role"
+		// TODO Audit Log logging
+		action {
+			event.interaction.respondEphemeral {
+				embed {
+					title = "Configuration: Logging"
+					field {
+						name = "Message Logs"
+						value = arguments.messageLogs.mention
+					}
+					field {
+						name = "Join/Leave Logs"
+						value = arguments.joinChannel.mention
+					}
+					footer {
+						text = "Configured by ${user.asUser().tag}"
+					}
+				}
+			}
+
+			LoggingConfigCollection().setConfig(
+				LoggingConfigData(
+					guild!!.id,
+					arguments.enableMessageLogs,
+					arguments.messageLogs.id,
+					arguments.enableJoinChannel,
+					arguments.joinChannel.id
+				)
+			)
 		}
+	}
+
+	// TODO Validate the presence of database values
+	unsafeSubCommand(::ClearArgs) {
+		name = "clear"
+		description = "Clear a config type"
 
-		/** The channel for creating support threads in. */
-		val supportChannel by optionalChannel {
-			name = "supportChannel"
-			description = "Your Support Channel"
+		initialResponse = InitialSlashCommandResponse.None
+
+		action {
+			when (arguments.config) {
+				ConfigType.MODERATION.name -> {
+					ModerationConfigCollection().clearConfig(guild!!.id)
+					event.interaction.respondEphemeral {
+						embed {
+							title = "Config cleared: Moderation"
+							footer {
+								text = "Config cleared by ${user.asUser().tag}"
+							}
+						}
+					}
+				}
+				ConfigType.LOGGING.name -> {
+					LoggingConfigCollection().clearConfig(guild!!.id)
+					event.interaction.respondEphemeral {
+						embed {
+							title = "Config cleared: Logging"
+							footer {
+								text = "Config cleared by ${user.asUser().tag}"
+							}
+						}
+					}
+				}
+				ConfigType.SUPPORT.name -> {
+					SupportConfigCollection().clearConfig(guild!!.id)
+					event.interaction.respondEphemeral {
+						embed {
+							title = "Config cleared: Support"
+							footer {
+								text = "Config cleared by ${user.asUser().tag}"
+							}
+						}
+					}
+				}
+				ConfigType.ALL.name -> {
+					ModerationConfigCollection().clearConfig(guild!!.id)
+					LoggingConfigCollection().clearConfig(guild!!.id)
+					SupportConfigCollection().clearConfig(guild!!.id)
+					event.interaction.respondEphemeral {
+						embed {
+							title = "Config cleared"
+							footer {
+								text = "Config cleared by ${user.asUser().tag}"
+							}
+						}
+					}
+				}
+			}
 		}
 	}
 }
+
+class SupportArgs : Arguments() {
+	val enable by boolean {
```

Comment:
Why is this an option? Wouldn't setting the config enable it? I think if we're having per-config booleans they should be set outside of the details of each config.

Reply:
No. Enabling and disabling features with a hard override is better and allows easier customisation. If a user sees an enable or disable, they'll know exactly what to do, if there isn't one they likely won't